legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/java/org/finos/legend/engine/persistence/components/e2e/unitemporal/UnitemporalDeltaWithBatchIdTest.java

```diff
@@ -0,0 +1,522 @@
+        dataSplitRanges.add(DataSplitRange.of(2, 2));
+
+        List<Map<String, Object>> expectedStatsList = new ArrayList<>();
+        Map<String, Object> expectedStatsSplit1 = createExpectedStatsMap(4, 0, 4, 0, 0);
+        Map<String, Object> expectedStatsSplit2 = createExpectedStatsMap(2, 0, 0, 2, 0);
+
+        expectedStatsList.add(expectedStatsSplit1);
+        expectedStatsList.add(expectedStatsSplit2);
+
+        executePlansAndVerifyResultsWithSpecifiedDataSplits(ingestMode, options, datasets, schema, expectedDataPass1, expectedStatsList, dataSplitRanges, fixedClock_2000_01_01, " order by \"batch_id_in\", \"id\"");
+
+        // ------------ Perform milestoning Pass2 ------------------------
+        ingestMode = ingestMode.withDeduplicationStrategy(FilterDuplicates.builder().build());
+        String dataPass2 = basePathForInput + "with_data_splits/staging_data_pass2.csv";
+        loadBasicStagingDataWithDataSplit(dataPass2);
+        String expectedDataPass2 = basePathForExpected + "with_data_splits/expected_pass2.csv";
+        // Execute plans and verify results
+        dataSplitRanges = new ArrayList<>();
+        dataSplitRanges.add(DataSplitRange.of(1, 1));
+        dataSplitRanges.add(DataSplitRange.of(2, 2));
+        dataSplitRanges.add(DataSplitRange.of(3, 3));
+
+        expectedStatsList = new ArrayList<>();
+        expectedStatsList.add(createExpectedStatsMap(4, 0, 1, 1, 0));
+        expectedStatsList.add(createExpectedStatsMap(1, 0, 0, 1, 0));
+        expectedStatsList.add(createExpectedStatsMap(1, 0, 0, 1, 0));
+
+        executePlansAndVerifyResultsWithSpecifiedDataSplits(ingestMode, options, Datasets.of(mainTable, stagingTable), schema, expectedDataPass2, expectedStatsList, dataSplitRanges, fixedClock_2000_01_01, " order by \"batch_id_in\", \"id\"");
+
+        // ------------ Perform milestoning Pass3 - Empty batch ------------------------
+        String dataPass3 = "data/empty_file.csv";
+        loadBasicStagingDataWithDataSplit(dataPass3);
+        String expectedDataPass3 = basePathForExpected + "with_data_splits/expected_pass3.csv";
+        // Execute plans and verify results
+        dataSplitRanges = new ArrayList<>();
+        expectedStatsList = new ArrayList<>();
+        expectedStatsList.add(createExpectedStatsMap(0, 0, 0, 0, 0));
+        executePlansAndVerifyResultsWithSpecifiedDataSplits(ingestMode, options, Datasets.of(mainTable, stagingTable), schema, expectedDataPass3, expectedStatsList, dataSplitRanges, fixedClock_2000_01_01, " order by \"batch_id_in\", \"id\"");
+    }
+
+    @Test
+    void testUniTemporalDeltaWithAllVersionGreaterThanAndStagingFilters() throws Exception
+    {
+        DatasetDefinition mainTable = TestUtils.getDefaultMainTable();
+        DatasetDefinition stagingDataset = DatasetDefinition.builder()
+                .group(testSchemaName)
+                .name(stagingTableName)
+                .schema(TestDedupAndVersioning.baseSchemaWithVersionAndBatch)
```

Comment:
I have already added the schema in TestUtils previously, we can just call **getDedupAndVersioningSchemaWithVersionAndBatch** here, then we dont need to duplicate the whole TestDedupAndVersioning file here